edisgo/network/electromobility.py

```diff
@@ -447,45 +470,312 @@ def _shorten_and_set_index(band):
+                "upper energy. Upper power band is therefore increased to avoid "
+                "infeasibilities arising from rounding errors."
+            )
+            self.flexibility_bands["upper_power"] += tol
+
+        # reduce lower energy band if there are cases where it is larger than upper
+        # energy band
         if (
             (
                 (
-                    lower_energy
-                    - upper_power * edisgo_obj.electromobility.eta_charging_points
+                    self.flexibility_bands["upper_energy"]
+                    - self.flexibility_bands["lower_energy"]
                 )
-                > 1e-6
+                < 0.0
             )
             .any()
             .any()
         ):
+            logger.debug(
+                "There are cases when lower energy band is larger than upper energy "
+                "band. Lower energy band is therefore reduced to avoid infeasibilities "
+                "arising from rounding errors."
+            )
+            self.flexibility_bands["lower_energy"] -= tol
+
+        # increase upper power, if there are cases where upper power is not sufficient
+        # to meet charged lower energy
+        if (
+            (
+                (
+                    self.flexibility_bands["lower_energy"].diff()
+                    - self.flexibility_bands["upper_power"] * efficiency / hourly_steps
+                )
+                > 0.0
+            )
+            .any()
+            .any()
+        ):
+            logger.debug(
+                "There are cases when upper power is not sufficient to meet charged "
+                "lower energy. Upper power band is therefore increased to avoid "
+                "infeasibilities arising from rounding errors."
+            )
+            self.flexibility_bands["upper_power"] += tol
+
+    def resample(self, freq: str = "15min"):
+        """
+        Resamples flexibility bands.
+
+        Parameters
+        ----------
+        freq : str or :pandas:`pandas.Timedelta<Timedelta>`, optional
+            Frequency that time series is resampled to. Offset aliases can be found
+            here:
+            https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases.
+            Default: '15min'.
+
+        """
+
+        flex_band = list(self.flexibility_bands.values())[0]
+        if flex_band.empty:
+            return
+
+        freq_orig = flex_band.index[1] - flex_band.index[0]
+        if not isinstance(freq, pd.Timedelta):
+            freq = pd.Timedelta(freq)
+        # in case of up-sampling, check if index is continuous and if new index fits
+        # into old index a discrete number of times
+        if freq < freq_orig:
+            check_index = pd.date_range(
+                start=flex_band.index.min(), end=flex_band.index.max(), freq=freq_orig
+            )
+            if not len(check_index) == len(flex_band.index):
```

Comment:
The check is sufficient because the check_index is set up right before and uses the minimum and maximum value of flex_band.index. So I would leave it like that.